Penteract/Assets/Scripts/HUDController.cpp

```diff
@@ -315,23 +333,27 @@ void HUDController::UpdateCommonSkill() {
 	}
 }
 
-void HUDController::UpdateFangCooldowns(GameObject* fangSkillCanvas) {
-	std::vector<GameObject*> children = fangSkillCanvas->GetChildren();
+
+void HUDController::UpdateFangCooldowns(GameObject* fangSkillCanvas, bool isMain) {
+	std::vector<GameObject*> skills = fangSkillCanvas->GetChildren();
 	int skill = FANG_SKILL_1;
-	for (std::vector<GameObject*>::iterator it = children.begin(); it != children.end(); ++it) {
-		std::vector<GameObject*> skills = (*it)->GetChildren();
-		for (std::vector<GameObject*>::iterator itSkills = skills.begin(); itSkills != skills.end(); ++itSkills) {
-			ComponentImage* image = (*itSkills)->GetComponent<ComponentImage>();
+	for (std::vector<GameObject*>::iterator it = skills.begin(); it != skills.end(); ++it) {
+
+		if ((*it)->GetChildren().size() > 0) {
+			ComponentImage* image = (*it)->GetChildren()[1]->GetComponent<ComponentImage>();
 			if (image) {
 				if (image->IsFill()) {
 					switch (skill) {
```

Comment:
There's no need for the switch. You can simply go: 

AbilityCoolDownEffectCheck(skill, fangSkillCanvas);
SetFillValue(cooldowns[skill]);

Reply:
Yup